### PR TITLE
Update  Storage Connection [MaxVolSupport]

### PR DESCRIPTION
**Pull request by michaelkad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing:

```
=== RUN   TestAccIBMPIInstancesDataSource_basic
--- PASS: TestAccIBMPIInstancesDataSource_basic (74.59s)
PASS

=== RUN   TestAccIBMPIInstanceDataSource_basic
--- PASS: TestAccIBMPIInstanceDataSource_basic (73.94s)
PASS

=== RUN   TestAccIBMPIInstanceStorageConnection
--- PASS: TestAccIBMPIInstanceStorageConnection (536.56s)
PASS
```

### CODE REVIEW

**Review by Diptipowervs — DISMISSED** [review]

File: website/docs/d/pi_instances.html.markdown [hunk 1]
```diff
@@ -42,36 +38,36 @@ Review the argument references that you can specify for your data source.
...
 
+  - `pin_policy` - (String) The pinning policy of the instance.
   - `placement_group_id`- (String) The ID of the placement group that the instance is a member.
   - `processors` - (Float) The number of processors that are allocated to the instance.
   - `proctype` - (String) The procurement type of the instance. Supported values are `shared` and `dedicated`.
   - `pvm_instance_id` - (String) The unique identifier of the instance.
-  - `shared_processor_pool`- (String) The name of the shared processor pool for the instance.
-  - `shared_processor_pool_id` - (String)  The ID of the shared processor pool for the instance.
   - `status` - (String) The status of the instance.
+  - `storage_connection` - (String) - The storage connection type for the instance.
```
Comment (ismirlia) [inline website/docs/d/pi_instances.html.markdown:69]: there's an extra dash between (String) and The

File: website/docs/d/pi_instance.html.markdown [hunk 2]
```diff
@@ -20,58 +18,59 @@ data "ibm_pi_instance" "ds_instance" {
...
   - `type` - (String) The type of the network.
+
+- `pin_policy` - (String) The pinning policy of the instance.
 - `placement_group_id`- (String) The ID of the placement group that the instance is a member.
 - `processors` - (Float) The number of processors that are allocated to the instance.
 - `proctype` - (String) The procurement type of the instance. Supported values are `shared` and `dedicated`.
 - `shared_processor_pool`- (String) The name of the shared processor pool for the instance.
 - `shared_processor_pool_id` - (String)  The ID of the shared processor pool for the instance.
 - `status` - (String) The status of the instance.
+- `storage_connection` - (String) - The storage connection type for the instance.
```
Comment (ismirlia) [inline website/docs/d/pi_instance.html.markdown:71]: there's an extra dash between (String) and The

Comment (michaelkad) [reply website/docs/d/pi_instance.html.markdown]: Updated

File: website/docs/d/pi_instances.html.markdown (hunk 1 above)
Comment (michaelkad) [reply website/docs/d/pi_instances.html.markdown]: Updated

**Comment by ismirlia** [comment]
Does the storageConnection not need to be added to the resource instance as an argument as well? It is currently an argument to the instance create command in the CLI.

```
--storage-connection string                 The storage connection type. Valid values are "vSCSI" or "maxVolumeSupport".
```

**Comment by ismirlia** [comment]
Oh the storage connection property is there in the resource, but not the new value. @michaelkad 

```
			helpers.PIInstanceStorageConnection: {
				Type:         schema.TypeString,
				Optional:     true,
				ValidateFunc: validate.ValidateAllowedStringValues([]string{"vSCSI"}),
				Description:  "Storage Connectivity Group for server deployment",
			},
```

File: website/docs/r/pi_instance.html.markdown [hunk 3]
```diff
@@ -100,7 +100,7 @@ Review the argument references that you can specify for your resource.
...
 - `pi_storage_pool_affinity` - (Optional, Boolean) Indicates if all volumes attached to the server must reside in the same storage pool. The default value is `true`. To attach data volumes from a different storage pool (mixed storage) set to `false` and use `pi_volume_attach` resource. Once set to `false`, cannot be set back to `true` unless all volumes attached reside in the same storage type and pool.
 - `pi_storage_type` - (Optional, String) - Storage type for server deployment; If storage type is not provided the storage type will default to `tier3`.
-- `pi_storage_connection` - (Optional, String) - Storage Connectivity Group (SCG) for server deployment. Only supported value is `vSCSI`.
+- `pi_storage_connection` - (Optional, String) - Storage Connectivity Group (SCG) for server deployment. Only supported values are `vSCSI`, `maxVolumeSupport`.
```
Comment (ismirlia) [inline website/docs/r/pi_instance.html.markdown:103]: Should probably say: "Support values are ..." instead of "Only supported values are..."

**Review by ismirlia — APPROVED** [review]

File: website/docs/r/pi_instance.html.markdown (hunk 3 above)
Comment (michaelkad) [inline website/docs/r/pi_instance.html.markdown:103]:
```suggestion
- `pi_storage_connection` - (Optional, String) - Storage Connectivity Group (SCG) for server deployment. Supported values are `vSCSI`, `maxVolumeSupport`.
```

**Comment by ismirlia** [comment]
Opened https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5667, closing this one.

**Pull request closed without merging**